react/src/reactTextMask.js

```diff
@@ -25,6 +25,33 @@ export default class MaskedInput extends React.PureComponent {
     this.initTextMask()
   }
 
+  componentDidUpdate(prevProps) {
+    // Getting props affecting value
+    const {value, pipe, mask, guide, placeholderChar, showMask} = this.props
+
+    // Сalculate that settings was changed:
+    // - `pipe` converting to string, to compare function content
+    // - `mask` converting to string, to compare values or function content
+    // - `keepCharPositions` exludes, because it affect only cursor position
+    const settings = {guide, placeholderChar, showMask}
+    const isPipeChanged = typeof pipe === 'function' && typeof prevProps.pipe === 'function' ?
+      pipe.toString() !== prevProps.pipe.toString() :
+      false
+    const isMaskChanged = mask.toString() !== prevProps.mask.toString()
```

Comment:
Do we need to guard against `mask` being `undefined` or `null`?

Reply:
I could, but it's required prop. If you think that we need to guard, than I'll do

Reply:
I just found out that it is marked as required in the proptypes, so that shouldn't be necessary.

Sorry, but it is a while since I worked on text-mask :-)